### PR TITLE
Worn gloves count as unarmed weapons when no weapon is held

### PR DESCRIPTION
**Pull request by CodeBandit**
#### Summary
SUMMARY: Features "Worn gloves count as unarmed weapons when no weapon is held"

#### Purpose of change
To facilitate the use of worn gloves as weapons when nothing is wielded since we are technically punching the enemies with them. This allows a unique quality to unarmed combat as we can now be "wielding" our weapons as soon as we drop whatever it is we are holding.

This PR is to facilitate this ability. Glove weapons and/or updates to existing gloves to have damage still need to happen. For the sake of not having dead code, and to prove out the feature, studded gloves have also been added.

#### Describe the solution
If the selected weapon is null (we aren't holding anything), then use the highest layer of gloves as our weapon instead. If it does not inherently have the unarmed combat flag, add it, then remove it later down the line.

This also converts cur_weapon from a reference to a pointer since reference variables cannot be reassigned. If not done this way, the gloves will be duped and held as a weapon instead.

Allowed the cestus to be able to be worn as an outer layer on hands, and still be used as an unarmed weapon.

#### Testing
Spawn light gloves, and punch debug monster. Make sure it trains unarmed, and the gloves can take damage.
Turn on force unarmed while wielding a cudgel, and repeat above.
Attack with a cudgel to make sure it still attacks with a held weapon.

### CODE REVIEW

**Comment by LaVeyanFiend** [comment]
Does this break martial arts that limit the unarmed weapons you're allowed to use, like Aikido and Judo?

File: data/json/items/armor/gloves.json [hunk 1]
```diff
@@ -948,5 +948,27 @@
...
+    "type": "ARMOR",
+    "name": { "str": "pair of studded gloves", "str_pl": "pairs of studded gloves" },
+    "description": "A pair of gloves with studded metal knuckles.",
+    "weight": "218 g",
+    "volume": "250 ml",
+    "price": 1100,
+    "price_postapoc": 250,
+    "bashing": 2,
+    "cutting": 1,
+    "material": [ "leather", "cotton" ],
```
Comment (Night-Pryanik) [inline data/json/items/armor/gloves.json:963]: Maybe also add steel as material? The gloves have scrap metal in their recipe, and description mentions metal.

File: src/melee.cpp [hunk 2]
```diff
@@ -422,17 +424,39 @@ void Character::melee_attack( Creature &t, bool allow_special, const matec_id &f
...
+                  worn_item.covers( bodypart_id( bp_hand_r ) ) ) ) {
+                if( cur_weapon->is_null() || ( worn_item.get_layer() >= cur_weapon->get_layer() ) ) {
+                    cur_weapon = &worn_item;
+                    found_glove = true;
+                }
+            }
+        }
+
+        // Treat all gloves as unarmed weapons.
+        if( found_glove && !cur_weapon->has_flag( flag_UNARMED_WEAPON ) ) {
```
Comment (Night-Pryanik) [inline src/melee.cpp:445]: You can remove the "has_flag" check as redundant. If an item already has this flag, "set_flag" below will simply do nothing.

File: data/json/items/armor/gloves.json [hunk 3]
```diff
@@ -948,5 +948,27 @@
...
+    "symbol": "[",
+    "looks_like": "gloves_leather",
+    "color": "light_gray",
+    "covers": [ "HANDS" ],
+    "coverage": 100,
+    "encumbrance": 2,
```
Comment (Night-Pryanik) [inline data/json/items/armor/gloves.json:969]: I'd increase encumbrance a bit to distinguish them from ordinary leather gloves.

File: src/melee.cpp (hunk 2 above)
Comment (AMurkin) [reply src/melee.cpp:445]: > You can remove the "has_flag" check as redundant. If an item already has this flag, "set_flag" below will simply do nothing.

It's `not has_flag`.

Comment (Night-Pryanik) [reply src/melee.cpp:445]: The logic is: if item doesn't have the flag, add it. The check could be omitted - we're adding flag anyways, and if there's already flag exist, adding flag just won't do anything.

Comment (AMurkin) [reply src/melee.cpp:445]: Right. Now I understand what you wrote before.

**Comment by Fris0uman** [comment]
>  If it does not inherently have the unarmed combat flag, add it, then remove it later down the line.

I don't understand the point of this?
" If it has the flag use it as a weapon, if it doesn't give it the flag and use it as a weapon" What's the point of the flag if everything has it?

**Comment by CptFail92** [comment]
Have you considered how this affects Taekwondo specifically? Ideally, the stats of one's shoes should apply. Barring that, Taekwondo becomes the only unarmed martial art that is--unrealistically--not affected by armor.

**Comment by CodeBandit** [comment]
I will not be adding the edge case of using shoes as weapons just to appease Taekwondo. That would be a seldom visited feature and adds unnecessary complexity.

**Comment by SirPendrak** [comment]
What about one-piece armor suits that have gloves/gauntlets incorporated like rm13, phase immersion suits, power armors etc?

**Comment by CodeBandit** [comment]
If it covers hands, then it can be added to them.

**Comment by Dacendeth** [comment]
what happens if im wearing like, 10 different pairs of gloves though?

**Comment by CodeBandit** [comment]
Added some unarmed weapons to be worn on hands as the outer layer to give the PR more relevance if merged.

**Comment by Rivet-the-Zombie** [comment]
>Added some unarmed weapons to be worn on hands

Great for the cestus, not so good for brass/steel/nail knuckles, which also [fill the hand with a grip](https://upload.wikimedia.org/wikipedia/commons/1/1d/Brass_knuckles.jpg), preventing you from holding much of anything else along with them.

**Comment by CodeBandit** [comment]
> Great for the cestus, not so good for brass/steel/nail knuckles, which also [fill the hand with a grip](https://upload.wikimedia.org/wikipedia/commons/1/1d/Brass_knuckles.jpg), preventing you from holding much of anything else along with them.

I erroneously thought they just wrapped around your fingers this whole time haha. I have updated it so only the cestus is allowed to be worn.

**Pull request merged**